src/sshagent/SSHAgent.h

```diff
@@ -32,14 +32,17 @@ class SSHAgent : public QObject
 
 public:
     static SSHAgent* instance();
-    static void init(MessageWidget* parent);
+    static void init(QObject* parent);
 
     const QString errorString() const;
     bool isAgentRunning() const;
     bool addIdentity(OpenSSHKey& key, quint32 lifetime = 0, bool confirm = false);
     bool removeIdentity(OpenSSHKey& key);
     void removeIdentityAtLock(const OpenSSHKey& key, const Uuid& uuid);
 
+signals:
+    void showMessage(const QString& message, MessageWidget::MessageType);
```

Comment:
Shouldn't this signal rather be called error or connectionFailed or something?

Reply:
Can't directly map it to MessageWidget slot if it doesn't include the second argument so I made it compatible. Can I somehow create a glue between argument incompatible signal/slot easily?

Reply:
I would simply use a separate signal for each type of event (if you have multiple) and then create a function that consumes this event and shows a message with the correct MessageType.